File: code/greeter.go
func Greet(runtime app.TaskRuntime, evtChan <-chan *app.TaskEvent, params ...interface{}) interface{} {
	r := runtime.(app.SignalTaskRuntime)
	if err := r.AcceptCall(); err != nil {
		return err
	}
	if err := r.PlayFile("http://media/greeting.mp3", false); err != nil {
		return err
	}
	for r.IsConnected() {
		evt := <-evtChan
		if r.IsPlayFileCompletedEvent(evt) {
			break
		}
	}
	return nil
}

File: code/greeter.lua
function Main()
   local err = AcceptCall()
   if err ~= nil then
      return
   end
   err = PlayFile("http://media/greeting.mp3")
   if err ~= nil then
      Disconnect()
      return
   end
   while IsConnected() do
      local evt = ReadInput(3600)
      if evt ~= nil and IsPlayFileCompletedEvent(evt) then
         break
      end
   end
end
